File: examples/java/PaperTrade.java
// Paper-trade differentiator demo.
//
// Run (with the binding classes + native library on the paths):
//   java --enable-native-access=ALL-UNNAMED -Dnative.lib.dir=<dir> \
//        -cp <wickra-exchange-classes> PaperTrade.java
import java.util.Map;
import org.wickra.exchange.Exchange;

public final class PaperTrade {
    public static void main(String[] args) {
        try (Exchange ex = Exchange.paper(Map.of("USDT", 100_000.0), 1, 5, 10)) {
            System.out.println("venue: " + ex.name());
            ex.setPrice("BTC/USDT", 20_000.0);

            Exchange.OrderInfo order = ex.placeMarket("BTC/USDT", Exchange.Side.BUY, 1.0);
            System.out.println("filled at " + order.averagePrice() + " (status " + order.status() + ")");

            System.out.println("  BTC free: " + ex.balance("BTC"));
            System.out.println("  USDT free: " + ex.balance("USDT"));
        }
    }
}

File: examples/go/paper_trade.go
// Paper-trade differentiator demo.
//
// Run from a module that requires github.com/wickra-lib/wickra-exchange-go:
//
//	go run paper_trade.go
package main

import (
	"fmt"

	wx "github.com/wickra-lib/wickra-exchange-go"
)

func main() {
	ex, err := wx.Paper(map[string]float64{"USDT": 100000}, 1, 5, 10)
	if err != nil {
		panic(err)
	}
	defer ex.Close()

	fmt.Println("venue:", ex.Name())
	if err := ex.SetPrice("BTC/USDT", 20000); err != nil {
		panic(err)
	}

	order, err := ex.PlaceMarket("BTC/USDT", wx.Buy, 1)
	if err != nil {
		panic(err)
	}
	fmt.Printf("filled at %v (filled=%v)\n", order.AveragePrice, order.IsFilled())

	btc, _ := ex.Balance("BTC")
	fmt.Printf("  BTC free: %v\n", btc)
}
